cli/command/stack/deploy_composefile.go

```diff
@@ -341,6 +350,12 @@ func deployServices(
 			if sendAuth {
 				createOpts.EncodedRegistryAuth = encodedAuth
 			}
+
+			// query registry if flag disabling it was not set
+			if !noResolveImage && versions.GreaterThanOrEqualTo(apiClient.ClientVersion(), "1.30") {
+				createOpts.QueryRegistry = true
+			}
```

Comment:
Instead of checking the version twice I think we should handle the version check in `runDeploy()` by setting the option accordingly (set the option to` true` for older versions). That way `noResolveImage` will always be correct inside `deployServices()` and we won't need to check version twice.